server/sandbox/create.go

```diff
@@ -0,0 +1,101 @@
+package sandbox
+
+import (
+	"context"
+)
+
+// Creator is the interface which reflects necessary steps in sandbox
+// creation. All methods have to be called in order to successfully creator a
+// pod sandbox.
+// TODO: add the necessary arguments to the methods
+type Creator interface {
```

Comment:
This interface is something that could be used to implement infra container based sandboxes i.e. our default code path or the libpod integration. 

This doesn't work for pinned namespace based containers or vm based containers, though. So, for those cases we may not even need a creator like interface to begin with.

So, we may need to specialize the name more than just Creatorr.

Reply:
So for VM based containers we cannot be more specific than the CRI (which already is an interface)? :thinking:

If so, them I'm wondering if another interface is the right approach or if we need something else to reduce the complexity… :grin:

Reply:
We may be able to get away with a single SandboxFactory function that takes the parameters and branches on the type to create, calling the constructor of whichever implementation of the Sandbox interface is needed, Instead of an entire Creator interface

Reply:
Okay, I tried to strip down the interface to a bare minimum and we could focus on two main high-level interfaces for now `Container` and `Sandbox`. WDYT? I think this way we could refactor into a plain new `pkg/{container,sandbox}` package.